gateway/src/test/java/io/zeebe/gateway/api/topology/TopologyTest.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.api.topology;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.zeebe.gateway.api.util.GatewayTest;
+import io.zeebe.gateway.impl.broker.cluster.BrokerClusterStateImpl;
+import io.zeebe.gateway.protocol.GatewayOuterClass.Partition.PartitionBrokerHealth;
+import io.zeebe.gateway.protocol.GatewayOuterClass.TopologyRequest;
+import org.junit.Test;
+
+public final class TopologyTest extends GatewayTest {
+
+  @Test
+  public void shouldResponseWithInitialUnhealthyPartitions() {
+    // when
+    final var response = client.topology(TopologyRequest.newBuilder().build());
+
+    // then
+    assertThat(response.getBrokersList())
+        .isNotEmpty()
+        .allSatisfy(
+            brokerInfo ->
+                assertThat(brokerInfo.getPartitionsList())
+                    .isNotEmpty()
+                    .allSatisfy(
+                        partitionPerBroker ->
+                            assertThat(partitionPerBroker.getHealth())
+                                .isEqualTo(PartitionBrokerHealth.UNHEALTHY)));
+  }
+
+  @Test
+  public void shouldUpdatePartitionHealthHealthy() {
+    // given
+    final var topology = (BrokerClusterStateImpl) brokerClient.getTopologyManager().getTopology();
+    topology.setPartitionHealthy(0, 1);
+
+    // when
+    final var response = client.topology(TopologyRequest.newBuilder().build());
+
+    // then
+    final var health = response.getBrokers(0).getPartitions(0).getHealth();
+    assertThat(health).isEqualTo(PartitionBrokerHealth.HEALTHY);
+  }
+
+  @Test
+  public void shouldUpdatePartitionHealth() {
+    // given
+    final var topology = (BrokerClusterStateImpl) brokerClient.getTopologyManager().getTopology();
+    topology.setPartitionHealthy(0, 1);
```

Comment:
You don't need this line in this test. I understand that it may seem so due to your implementation, but I'd rather test behaviour. So just setting 1 partition to unhealthy should be enough for this test.

Reply:
Done